ENIGMAsystem/SHELL/Universal_System/Extensions/FileManip/POSIX/fileman.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+
+ MIT License
+ 
+ Copyright © 2020 Samuel Venable
+ 
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+ 
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+ 
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+ 
+*/
+
+#include "../fileman.h"
+#include <unistd.h>
+#include <cstdlib>

```

Comment:
C++ headers should be included before C headers, because they often include C headers themselves. Switch these two lines.

Reply:
unistd.h is not a part of the C standard, it is a posix header.

Reply:
Well the general rule is that C++ headers are included before C headers. If you start doing this now, other people get confused and will end up including more C headers there. That's why it's important to be consistent across all source files about the include order.

Reply:
Also, it doesn't matter if `unistd.h` is part of the C standard, it's still a C header indicated by the `.h` extension.

Reply:
Thank you for the clarification. I just wanted to double check. Does this also apply to Windows headers like windows.h and shlobj.h? I guess it's safe to assume this isnt the only file you want changed, but pretty much all of them lol will do

Reply:
Yeah, the Windows header is a C header. Also, yeah, it's easier to do it now rather than later.